Add some of the introductory material (#8)
space-ros/docs: source/Introduction/How-SpaceROS-Differs.rst

> How Space ROS Differs from ROS 2

Reviewer: Again this file might be renamed `How-Space-ROS-Differs.rst` to match the spelling conventions for Space ROS.

Author: Yes. Changed the filename.

================================

To date, work has focused on establishing the foundation for Space ROS, including builds and infrastructure, code analysis tools, requirements tools and processes, and space-specific functionality.
These efforts extend the foundation provided by ROS 2.
The following sections list the differences between Space ROS and the standard ROS 2 distributions.

Builds and Infrastructure
-------------------------
..

>   Space ROS builds include a subset of ROS 2 packages.

Reviewer: Comment directive. Intentional or should it be removed?

Author: Intentional, but now removed.

> There are several new infrastructure elements for Space ROS, including  information resources and regular builds provided to the Space ROS community.

Reviewer: Small fix: a double space.

```suggestion
There are several new infrastructure elements for Space ROS, including information resources and regular builds provided to the Space ROS community.
```

Author: Fixed.

.. list-table::
   :widths: 25 75
   :header-rows: 1

   * - Item
     - Description
   * - Website

>      - The `Space ROS website <https://space-ros.github.io/spaceros.org/>`_ has high-level information about Space ROS as well as a searchable set of related articles and videos.

Reviewer: ```suggestion
     - The `Space ROS website <http://space.ros.org/>`_ has high-level information about Space ROS as well as a searchable set of related articles and videos.
```

Now we can use the official url!

Author: Yay, updated!

   * - Documentation site
     - The Space ROS documentation site (this site) contains information about Space ROS, without duplicating the standard ROS 2 documentation.
       For standard ROS 2 topics, one can refer to the `ROS 2 Humble documentation <https://docs.ros.org/en/humble/index.html>`_.

>    * - Github organization

Reviewer: I hate that I still care about this but it is GitHub rather than Github.

```suggestion
   * - GitHub organization
```

Author: Fixed.

>      - There is a `github organization <https://github.com/space-ros>`_ for all of the Space ROS projects.

Reviewer: Since Space ROS is largely composed of software from ROS 2 and ament, several Space ROS projects (like ament_cobra) are hosted in other organizations.

```suggestion
     - There is a `github organization <https://github.com/space-ros>`_ for many of the Space ROS projects.
```

Author: True, fixed.

   * - Docker images
     - Pre-built Docker images are available from the `OSRF Space ROS page <https://hub.docker.com/r/osrf/space-ros>`_ on Docker Hub.
   * - Earthly builds

>      - Space ROS builds use the `Earthly build tool <https://earthly.dev/>`_.

Reviewer: ```suggestion
     - Space ROS builds use the `Earthly CI/CD tool <https://earthly.dev/>`_.
```

"build tool" has a very precise definition within ROS (see http://design.ros2.org/articles/build_tool.html) which Earthly does not meet so I propose rephrasing this slightly.

Author: Fixed.
